jmh-benchmarks/src/main/java/org/apache/kafka/jmh/assignor/TargetAssignmentBuilderBenchmark.java

```diff
@@ -0,0 +1,218 @@
+import org.apache.kafka.coordinator.group.consumer.SubscribedTopicMetadata;
+import org.apache.kafka.coordinator.group.consumer.TargetAssignmentBuilder;
+import org.apache.kafka.coordinator.group.consumer.TopicMetadata;
+import org.apache.kafka.coordinator.group.consumer.VersionedMetadata;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@Fork(value = 1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class TargetAssignmentBuilderBenchmark {
+
+    @Param({"100", "500", "1000", "5000", "10000"})
+    private int memberCount;
+
+    @Param({"5", "10", "50"})
+    private int partitionsToMemberRatio;
+
+    @Param({"10", "100", "1000"})
+    private int topicCount;
+
+    @Param({"true", "false"})
+    private boolean isRackAware;
+
+    String groupId = "benchmark-group";
+
+    private static final int groupEpoch = 0;
+
+    private PartitionAssignor partitionAssignor;
+
+    private Map<String, TopicMetadata> subscriptionMetadata = Collections.emptyMap();
+
+    private TargetAssignmentBuilder targetAssignmentBuilder;
+
+    private AssignmentSpec assignmentSpec;
+
+    private static final int numberOfRacks = 3;
+
+    private final List<String> allTopicNames = new ArrayList<>(topicCount);
+
+    private final List<Uuid> allTopicIds = new ArrayList<>(topicCount);
+
+    @Setup(Level.Trial)
+    public void setup() {
+        // For this benchmark we will use the Uniform Assignor
+        // and a group that has a homogeneous subscription model.
+        partitionAssignor = new UniformAssignor();
+        subscriptionMetadata = generateMockSubscriptionMetadata();
+        Map<String, ConsumerGroupMember> members = generateMockMembers();
+        Map<String, Assignment> existingTargetAssignment = generateMockInitialTargetAssignment();
+
+        // Add a new member to trigger a rebalance.
+        Set<String> subscribedTopics = new HashSet<>(subscriptionMetadata.keySet());
+        String rackId = isRackAware ? "rack" + (memberCount - 1) % numberOfRacks : "";
```

Comment:
nit: Should we extract it into an helper method too?

Reply:
It is already defined. Let's use it.

Reply:
It is defined with the Optional return value